From 696619729ff7145a9f957932d7c57aeb87061e21 Mon Sep 17 00:00:00 2001
From: Georgy Shelkovy <rekgrpth@gmail.com>
Date: Mon, 23 Dec 2024 08:51:39 +0500
Subject: [PATCH] Fix compatibility with pg18

Upstream commit postgres/postgres@5983a4c changed the "TupleDescData" struct. Use "TupleDescAttr" macro to access in all versions.
---
 plpgsql_debugger.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/plpgsql_debugger.c b/plpgsql_debugger.c
index 381b8c6..85ad230 100644
--- a/plpgsql_debugger.c
+++ b/plpgsql_debugger.c
@@ -681,11 +681,7 @@ print_rec(const PLpgSQL_execstate *frame, const char *var_name, int lineno,
 	{
 		char * extval = SPI_getvalue( tuple, rec_tupdesc, attNo + 1 );
 
-#if (PG_VERSION_NUM >= 110000)
-		dbg_send( "v:%s.%s:%s\n", var_name, NameStr( rec_tupdesc->attrs[attNo].attname ), extval ? extval : "NULL" );
-#else
-		dbg_send( "v:%s.%s:%s\n", var_name, NameStr( rec_tupdesc->attrs[attNo]->attname ), extval ? extval : "NULL" );
-#endif
+		dbg_send( "v:%s.%s:%s\n", var_name, NameStr( TupleDescAttr(rec_tupdesc, attNo)->attname ), extval ? extval : "NULL" );
 
 		if( extval )
 			pfree( extval );
